pkg/util/tls/tls.go

```diff
@@ -74,12 +72,27 @@ func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
 	return config, nil
 }
 
+type ClientConfigWithEnabled struct {
+	ClientConfig `yaml:",inline"`
+	Enabled      bool `yaml:"tls_enabled"`
```

Comment:
I expected to see `Enabled` flag in `grpcclient.Config`, and then using that instead of current `ClientConfigWithEnabled`. WDYT? (It may require passing `tlsEnabled` flag to `GetGRPCDialOptions`)

Reply:
I actually have implemented it that way the first time around.

It ended up being quite tedious in terms of duplicated work and passing the boolean around. 

Esp. in places where there is not a full `grpcclient.Config` used, like here:

https://github.com/cortexproject/cortex/blob/342e17c592e6c63c3d01227732a7ea0ebdb27111/pkg/querier/querier.go#L63
or there:
https://github.com/cortexproject/cortex/blob/342e17c592e6c63c3d01227732a7ea0ebdb27111/pkg/querier/blocks_store_replicated_set.go#L40

Wdyt if we maybe call it `tls.GRPCClientConfig` instead of `tls.ClientConfigWithEnabled` as this is more accurate in the way we use it